main.py

```diff
@@ -8,10 +8,22 @@
 
 @app.route('/', methods=["POST", "GET"])
 def main():
-    #pull_request = PullRequest(request.get_json())
-    pull_request = PullRequest()
-    pull_request.set_issue_url('https://api.github.com/repos/dimagi/commcare-android/issues/1858')
-    return str(create_status_json(pull_request, REQUIRED_LABELS_ANY, REQUIRED_LABELS_ALL, BANNED_LABELS))
+    #event_json = request.get_json()
+    with open('./tests/pr_event_no_labels.json') as json_file:
+        event_json = json.load(json_file)
+
+    if event_warrants_label_check(event_json):
+        pull_request = PullRequest(event_json)
+        status_code = pull_request.post_status(
+            create_status_json(pull_request, REQUIRED_LABELS_ANY, REQUIRED_LABELS_ALL, BANNED_LABELS))
+        return str(status_code)
+    else:
+        return 'No label check needed'
```

Comment:
should this return a green checkmark in github instead? Or do you know how we would mark a PR as having a check that is not needed?

Reply:
I don't think either is necessary (and setting a green checkmark would be incorrect). The point is that this represents the condition where the PR action that triggered the call to our webhook is just not of any interest to our purpose. If the action is something other than 1 of `["opened", "reopened", "labeled", "unlabeled"]`, then we can be certain that the status has already been set based on the last of those events that was relevant (presumably we're guaranteed that at least 'opened' has happened before, if we're now receiving something else), and thus nothing needs to be re-evaluated right now.